rl4co/models/zoo/nargnn/encoder.py

```diff
@@ -71,6 +71,7 @@ def _make_heatmaps(self, batch_graph: Batch) -> Tensor:  # type: ignore
             device=device,
             dtype=graphs[0].edge_attr.dtype,
         )
+        heatmaps_logits.fill_(-10)
```

Comment:
Why do we do this? Is it because we use the decoding strategy?

Reply:
In the sparsification stage, some long edges are ruled out from the graph to reduce computational costs of the GNN. 
In the original implementation the heuristic value of these excluded edges are set to a small value. But here we are using logits, and 0 corresponds to 1 after exponentiation. This could "spread out" the heuristics and increase the probability of selecting those long edges, which causes the performance issue.
The solution is to set the value to a negative number, and I think -10 is small enough, as the range of the GNN output is $[0,10]$.

Reply:
Oh, I see the reason; the logits are in [-10, +10], due to line 58. So, the logits for masked edges should be -10 to inactivate them.

Reply:
BTW why is the logit in that range?

Reply:
@fedebotu BTW, why we use sigmoid for the logit values? Wouldn't it be better to use tanh to avoid this postprocessing?

Reply:
I was thinking of keeping the network structure while implementing DeepACO in rl4co, so I didn't pay much attention to the migration from probs to logits and change the output activation function. 
I'm open to suggestions and I believe we can find a better plan for it. 😄

Reply:
Can we avoid this by taking logarithms of the heuristics we did for the pheromone [here](https://github.com/ai4co/rl4co/pull/170/files#diff-826642af91d93eb12e98537d3c99bfdb6f472be68518aba0dcab72b1c412791dR127)?

i.e., like this: 
`heatmaps_logits = (self.alpha * torch.log(self.pheromone) + self.beta * torch.log(self.heuristic))`

Would this cause some side effects?

Reply:
I tried to replace it with the original output layer as in DeepACO (`heatmaps_logits` = $\log\sigma(x)$ ), but the performance is worse than the current version (`heatmaps_logits` = $10\sigma(x)$ ). So I think there might be some advantages with the current version. 🤔

Reply:
A small note on this: why do we have a scaled sigmoid $10 \sigma(x)$ on the last layer in the first place? Is it for regularization? Asking because we have the [decoding strategy](https://github.com/ai4co/rl4co/blob/c546234360e4a5e772aa8a3ae8e58e85d0698492/rl4co/utils/decoding.py#L188) that processes the logits afterwards anyways

Reply:
I think it's about the scale of the logits, which is related to the sharpness of the probability.
Maybe we can avoid using $10\sigma(x)$ by adjusting the temperature?

Reply:
@hyeok9855 yes, that is also an option. $\tau$ set to 0.1 should have a similar effect

Reply:
Great idea, I'll use temperature instead.